### PR TITLE
Story 24

### PR DESCRIPTION
**Pull request by agallant121**
Complete user story 24

### CODE REVIEW

File: app/controllers/cart_controller.rb [hunk 1]
```diff
@@ -22,19 +22,34 @@ def remove_item
...
+    adjust_quantity
+  end
+
   private
 
   def require_not_admin
       render file: "/public/404" if current_admin
   end
 
-  # def increment_decrement
```
Comment (agallant121) [inline app/controllers/cart_controller.rb:31]: Deleted commented out code left by instructors

File: app/controllers/cart_controller.rb [hunk 2]
```diff
@@ -22,19 +22,34 @@ def remove_item
...
+  end
+
+  def adjust_quantity
+    if params[:quantity] == "add" && item.inventory > session[:cart][params[:item_id]]
+      session[:cart][params[:item_id]] += 1
+      redirect_to '/cart'
+    elsif params[:quantity] == "add"
+      flash[:error] = "Out of Stock"
+      redirect_to '/cart'
+    elsif params[:quantity] == "subtract" && session[:cart][params[:item_id]] > 1
```
Comment (agallant121) [inline app/controllers/cart_controller.rb:46]: Add the same functionality, only reversed, to decrement items quantity in cart

File: app/controllers/cart_controller.rb [hunk 3]
```diff
@@ -22,19 +22,34 @@ def remove_item
...
+      session[:cart][params[:item_id]] -= 1
```
Comment (agallant121) [inline app/controllers/cart_controller.rb:47]: Quantity greater than one, then decrement by one.

File: app/controllers/cart_controller.rb [hunk 4]
```diff
@@ -22,19 +22,34 @@ def remove_item
...
+      redirect_to '/cart'
+    else params[:quantity] == "subtract" && session[:cart][params[:item_id]] == 1
```
Comment (agallant121) [inline app/controllers/cart_controller.rb:49]: Quantity equal to one, then remove item from cart using helper method from above in the controller.

File: app/views/cart/show.html.erb [hunk 5]
```diff
@@ -30,8 +30,9 @@
       </td>
       <td>
         <center>
-          <p><%=quantity%></p>
-          <p><%= link_to "Remove", "/cart/#{item.id}",method: :delete %></p>
+          <p id = "item-quantity-<%=item.id%>"><%=quantity%></p>
+          <p><%= link_to "+", "/cart/#{item.id}/add", method: :patch %> or <%= link_to "-", "/cart/#{item.id}/subtract", method: :patch %></p>
```
Comment (agallant121) [inline app/views/cart/show.html.erb:34]: Add '-' link to decrement quantity of item in cart

File: config/routes.rb [hunk 6]
```diff
@@ -30,6 +30,8 @@
   get "/cart", to: "cart#show"
   delete "/cart", to: "cart#empty"
   delete "/cart/:item_id", to: "cart#remove_item"
+  patch "/cart/:item_id/:quantity", to: "cart#edit_quantity"
```
Comment (agallant121) [inline config/routes.rb:33]: Uses the same route to go to the same action which both increments and decrements

File: spec/features/cart/increment_cart_spec.rb [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+      within "#cart-item-#{@paper.id}" do
+        click_on '+'
+      end
+
+      expect(current_path).to eq('/cart')
+      expect(page).to have_content("Out of Stock")
+    end
+
+    it 'I can decrement the item until it is removed from my cart' do
```
Comment (agallant121) [inline spec/features/cart/increment_cart_spec.rb:62]: Just walks through the steps of decrementing the items quantity until it is at zero which then removes the item from the cart.

**Review by meghanstovall — APPROVED** [review]

**Pull request merged**